### PR TITLE
Creating plugin passthrough enhancement proposal

### LINKED ISSUE #27: Plugins Passthrough Information
Plugins must have a method of requesting extra information to be added to the plugin. The CLI for transformation will also need to be able to expose these as CLI flags

### PR DESCRIPTION
**Pull request by shawn-hurley**
Initial enhancement for creating plugin pass-through information. 

fixes #27

### CODE REVIEW

File: enhancements/crane-2.0/plugin-passthrough-info/README.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+We will provide a call via standard-in for a plugin to tell the `BinaryPlugin` type that it has specific optional fields as strings. If the plugin is expecting some numeric type, they will have to validate it on their side. A particular plugin error is created when this validation fails to communicate this type of failure. 
+
+We will create a Standard-In call, `Metadata`, which when called will cause a binary plugin to print out the `PluginMetadata` type. The Plugin library will be responsible for this and will be responsible for determining the versions of Response and Request's that it can handle. We will also add a version for the plugin itself and a name of the plugin. It will also return with a set of optional information that could be added.
+
+### User Stories [optional]
+
+#### Story 1
+
+As a user, I would like to change the registry information for a set of registries like `crane transform --optional="--registry-replace quay.io/testing->registry.redhat.io/testing"`.
```
Comment (alaypatel07) [inline enhancements/crane-2.0/plugin-passthrough-info/README.md:73]: NIT: I wonder if `--optional` really suggest this is an flag for the plugin. wonder if we can rename that flag `--plugin-args` or something?

Comment (shawn-hurley) [reply enhancements/crane-2.0/plugin-passthrough-info/README.md:73]: +1, I am wondering if that UX is even something we want, I would love for us to be able to do this nicer

**Review by alaypatel07 — APPROVED** [review]
+1, overall this makes a lot of sense to me.

/lgtm

File: enhancements/crane-2.0/plugin-passthrough-info/README.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+## Summary 
+
+A plugin developer wants the ability to ask for more information to more fully create the correct transformations for
+a kubernetes manifest.
+As plugins **MUST NOT** have access to clusters, a plugin user must give the information that in some current usage patterns is determined from a cluster. 
+
+## Motivation
+
+As a user create and use plugins, they will need to add extra information during the creation of transformations so
```
Comment (sseago) [inline enhancements/crane-2.0/plugin-passthrough-info/README.md:45]: NIT: "As a user creates and uses"

**Review by sseago — APPROVED** [review]
LGTM, just a minor wording issue under motivation.

**Pull request merged**